File: selection_sort.go
package main

import "fmt"

type int_array []int

func (arr int_array) selection_sort() {
	size := len(arr)
	for i := 0; i < size-1; i++ {
		for j := i + 1; j < size; j++ {
			if arr[j] < arr[i] {
				arr[i], arr[j] = arr[j], arr[i]
			}
		}
	}
}

func main() {
	var size int
	fmt.Print("Enter the size of the array: ")
	fmt.Scanf("%d", &size)
	fmt.Printf("\n")

	arr := make(int_array, size)
	fmt.Println("Enter the elements of the array:")
	for i := 0; i < size; i++ {
		fmt.Scanf("%d", &arr[i])
	}
	fmt.Printf("\n")

	arr.selection_sort()

	fmt.Println("Sorted array is:")
	for _, n := range arr {
		fmt.Print(n, " ")
	}
	fmt.Printf("\n")
}

File: selectionsort.java
import java.util.*;
public class selectionsort {
    public static void main(String[] args) {
     Scanner sc = new Scanner(System.in);
     int ar[]=new int[8];
     System.out.println("ENTER THE ELEMENTS");
     for(int i=0;i<8;i++)
     {
        ar[i]=sc.nextInt();
     }
     int min;min=0;
     for(int i=0;i<7;i++)
     {
        min=i;   
        
        for(int j=i+1;j<6;j++)
        {
            if(ar[j]<ar[min])
            {
            min=j;
            }
        }
    
    int temp=ar[min];
    ar[min]=ar[i];
    ar[i]=temp;
     }
     for(int i=0;i<8;i++)
     {
        System.out.print(ar[i]+ " ");
     }
    }

}
